src/test/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/end2end/AdminPageEnd2EndTest.java

```diff
@@ -0,0 +1,109 @@
+    }
+
+    private WebDriver webDriver;
+
+
+    @Rule
+    public WireMockRule mockOAuth2Provider = new WireMockRule(wireMockConfig()
+            .port(8077)
+            .extensions(new ResponseTemplateTransformer(true)));
+
+    @Before
+    public void setUp() {
+        // Setup ChromeDriver (aided by the WebDriverManager)
+        WebDriverManager.chromedriver().setup();
+        ChromeOptions options = new ChromeOptions();
+        //Run browser in headless mode for CI
+        options.addArguments("--headless");
+        webDriver = new ChromeDriver(options);
+
+        // Configure mock oauth endpoints
+        mockOAuth2Provider.stubFor(get(urlPathEqualTo("/favicon.ico")).willReturn(notFound()));
+        mockOAuth2Provider.stubFor(get(urlPathMatching("/oauth/authorize?.*"))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "text/html")
+                        .withBodyFile("mockLogin.html")));
+        mockOAuth2Provider.stubFor(post(urlPathEqualTo("/login"))
+                .willReturn(temporaryRedirect("{{formData request.body 'form' urlDecode=true}}http://localhost:8080/login/oauth2/code/wiremock?code={{{randomValue length=30 type='ALPHANUMERIC'}}}&state={{{form.state}}}")));
+        mockOAuth2Provider.stubFor(post(urlPathEqualTo("/oauth/token"))
+                .willReturn(okJson("{\"token_type\": \"Bearer\",\"access_token\":\"{{randomValue length=20 type='ALPHANUMERIC'}}\"}")));
+        mockOAuth2Provider.stubFor(get(urlPathEqualTo("/userinfo"))
+                .willReturn(okJson("{\"sub\":\"my-id\",\"email\":\"joe@ucsb.edu\", \"hd\":\"ucsb.edu\", \"name\":\"Joe\", \"given_name\":\"Joe\", \"family_name\":\"Gaucho\"}")));
+    }
+
+    //Close the browser after tests are done
+    @After
+    public void reset() {
+        if (webDriver != null) {
+            webDriver.close();
+        }
+    }
+
+
+    @Test
+    public void runUserFlowEnd2EndTestWithAuthentication() {
+        // Navigate to login page
+        webDriver.get("http://localhost:8080/oauth2/authorization/wiremock");
+        // Make sure Spring redirected us to the right place
+        assert(webDriver.getCurrentUrl()).startsWith("http://localhost:8077/oauth/authorize");
+        webDriver.findElement(By.id("submit")).click();
+        // Verify login was successful (and did not result in /error path)
+        assert(webDriver.getCurrentUrl().equalsIgnoreCase("http://localhost:8080/"));
+        // Navigate to the admin page
+        webDriver.get("http://localhost:8080/admin");
+        // Verify that we are at the correct url
+        assert(webDriver.getCurrentUrl().equalsIgnoreCase("http://localhost:8080/admin"));
+        // Find the form input and create an admin
+        webDriver.findElement(By.id("selenium-email-input")).sendKeys("testAdmin@ucsb.edu");
+        webDriver.findElement(By.id("selenium-admin-add")).click();
```

Comment:
if you use `data-selenium="admin-add"` then maybe search by CSS selector, and the syntax `[data-selenium=admin-add]` or something like that; you'll figure it out.